File: 1398/1398.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var customersWhoBoughtProductsAAnd = function(root) {
    if (!root) return 0;
    return 1 + Math.max(customersWhoBoughtProductsAAnd(root.left), customersWhoBoughtProductsAAnd(root.right));
};

File: 1398/1398.go
func customersWhoBoughtProductsAAnd(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := customersWhoBoughtProductsAAnd(root.Left)
	r := customersWhoBoughtProductsAAnd(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}